File: Ems.App/ClientApp/src/app/shared/models/admincourse.model.ts
export class AdminCourse {
  courseId: string;
  courseName: string;
  subtitle: string;
  description: string;
  startDate: string;
  endDate: string;
  startTime: string;
  endTime: string;
  }

export class AdminChapter {
  chapterId: string;
  courseId: string;
  title: string;
  contents: AdminChapterContent[]
}

export class AdminChapterContent {
  contentId: string;
  chapterId: string;
  contentTitle: string;
  body: string;
}
export class AdminQuestion {
  questionId: string;
  questionText: string;
  options: AdminOption[]
}
export class AdminOption {
  optionId: string;
  optionText: string;
  isCorrect: boolean;
}

File: Ems.App/Models/AdminCourseModel.cs
using Ems.Data.Entities;

namespace Ems.App.Models
{
    public class AdminCourseModel
    {
        public Guid courseId { get; set; }
        public string courseName { get; set; }
        public string subtitle { get; set; }
        public string description { get; set; }
        public DateOnly startDate { get; set; }
        public DateOnly endDate { get; set; }
        public TimeOnly startTime { get; set; }
        public TimeOnly endTime { get; set; }

    }

    public class AdminChapterModel
    {
        public Guid chapterId { get; set; }
        public Guid courseId { get; set; }
        public string title { get; set; }
    }

    public partial class AdminChapterContentModel
    {
        public Guid contentId { get; set; }
        public Guid chapterId { get; set; }
        public string contentTitle { get; set; }
        public string body { get; set; }
    }
}
